core/random.rbs

```diff
@@ -81,8 +134,9 @@ class Random < Object
   # (`-`) and add (`+`)methods, or rand will raise an ArgumentError.
   #
   def rand: () -> Float
-          | (Integer | ::Range[Integer] max) -> Integer
+          | (Integer | ::Range[Integer] | Numeric max) -> Integer
           | (Float | ::Range[Float] max) -> Float
+          | (::Range[Numeric]) -> Numeric
```

Comment:
Yeah, the type we want would be `[T <: Numeric] (Range[T]) -> T`, but RBS currently doesn't allow writing it.

I think there are two options. `[T] (Range[T]) -> T` or `(Range[Numeric]) -> Numeric`.  The first option would make sense in my opinion, because testing the code will uncover the problem anyway.